ui.apps/src/main/content/jcr_root/apps/venia/components/commerce/productcarousel/v1/productcarousel/clientlibs/js/product-carousel.js

```diff
@@ -0,0 +1,131 @@
+/*******************************************************************************
+ *
+ *    Copyright 2019 Adobe. All rights reserved.
+ *    This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License. You may obtain a copy
+ *    of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software distributed under
+ *    the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ *    OF ANY KIND, either express or implied. See the License for the specific language
+ *    governing permissions and limitations under the License.
+ *
+ ******************************************************************************/
+
+(function() {
+
+  /**
+   *  Product Carousel Component  
+   */
+  class ProductCarousel {
+
+    // Initialise the variables
+    static selectors = {
+
+      self: "[data-comp-is=productcarousel]",
+      btnPrev: "[data-carousel-action=prev]",
+      btnNext: "[data-carousel-action=next]"
+
+    }
+
+    constructor(rootElement) {
+      this._speed = 300;
```

Comment:
It may not be valid for this ticket but we can think of making delay and speed configurable, with some default values.

Reply:
This is not in the scope for this ticket